### ISSUE TITLE
Relational recurrent neural networks

### ISSUE THREAD

**Issue by icoxfog417** [arXivTimes/arXivTimes#784]
## 一言でいうと

メモリネットワークを改良し、LSTMに組み込んだ研究。メモリは行列で表現され、入力を含めた各行に対しクエリ(Q)とキー(K)・値(V)を算出し、クエリとキーの近さで重み(=Attention)を計算し値に乗じて更新を行う(softmax(QK^T)V)。言語モデルなどの教師ありと、強化学習で効果を確認。

![image](https://user-images.githubusercontent.com/544269/41138589-429de72a-6b1d-11e8-833c-8c87924de29b.png)

### 論文リンク

https://arxiv.org/abs/1806.01822

### 著者/所属機関

Adam Santoro, Ryan Faulkner, David Raposo, Jack Rae, Mike Chrzanowski, Theophane Weber, Daan Wierstra, Oriol Vinyals, Razvan Pascanu, Timothy Lillicrap

* DeepMind
* CoMPLEX, Computer Science, University College London

### 投稿日付(yyyy/MM/dd)

2018/6/5

**Comment by icoxfog417** [comment]
Sonnetベースの実装あり(使われていたのか。。。)
https://github.com/deepmind/sonnet/blob/master/sonnet/python/modules/relational_memory.py
